### PR TITLE
Add ability to pass "valid" as a prop to the Textfield component

### LINKED ISSUE #332: Textfield does not implement `valid` prop
I'm trying to pass a prop `valid` to `<Textfield />`, as described [here](https://material.io/components/web/catalog/input-controls/text-fields/#mdctextfield-api)

Doesn't seem to do anything. 

I would expect it to add the appropriate classes, for example `.mdc-textfield--invalid` to the top level div.

### PR DESCRIPTION
**Pull request by mellogarrett**
Fixes #332

### CODE REVIEW

File: Textfield/Textfield.jsx [hunk 1]
```diff
@@ -115,11 +119,26 @@ class Textfield extends Component {
   }
 
   componentDidMount() {
+    this._setInvalid(defaultProps, this.props, this.MDComponent);
     this.setState({
       showFloatingLabel: true
     });
   }
 
+  _setInvalid(oldprops, newprops, textfield) {
```
Comment (prateekbh) [inline Textfield/Textfield.jsx]: this does not need to be a part of the class

**Comment by mellogarrett** [comment]
Hmm. `textfield` is `undefined` when `setInvalid` is called from `componentDidMount`...

**Comment by mellogarrett** [comment]
Okay check it out now. It's been tested and is working for both starting with `valid: false` and also setting it to `false` after initial mount.

File: Textfield/Textfield.jsx [hunk 2]
```diff
@@ -119,24 +120,13 @@ class Textfield extends Component {
...
-      "valid" in newprops &&
-      oldprops.valid !== newprops.valid
-    ) {
-      textfield.valid = newprops.valid;
-    }
-  }
-
   componentWillUpdate(nextProps) {
-    this._setInvalid(this.props, nextProps, this.MDComponent);
+    setValid(this.props, nextProps, this.MDComponent);
```
Comment (prateekbh) [inline Textfield/Textfield.jsx]: This class doesn't have any `MDComponent`, `TextfieldInput` has. Shouldn't this be there?

Comment (mellogarrett) [reply Textfield/Textfield.jsx]: Yes, that is correct

**Comment by prateekbh** [comment]
wow prettier is failing! i'll have a look

File: Textfield/Textfield.jsx [hunk 3]
```diff
@@ -83,9 +91,9 @@ class TextfieldInput extends MaterialComponent {
           />
         )}
         {props.label &&
-        this.state.showFloatingLabel && (
-          <Label for={props.id}>{props.label}</Label>
-        )}
+          this.state.showFloatingLabel && (
+            <Label for={props.id}>{props.label}</Label>
+          )}
```
Comment (filoozom) [inline Textfield/Textfield.jsx]: I assume this is breaking ```prettier```.
Can you run ```npm run prettier:write``` and push?

Comment (mellogarrett) [reply Textfield/Textfield.jsx]: `npm run prettier:write` did not result in any committable changes

Comment (filoozom) [reply Textfield/Textfield.jsx]: Why did this change anyway? Was it ```prettier``` or did you change this by hand?

Comment (mellogarrett) [reply Textfield/Textfield.jsx]: Prettier is changing this. Should I turn off Prettier and push the old version?

Comment (filoozom) [reply Textfield/Textfield.jsx]: For debugging purposes:
- what ```prettier``` version do you have?
- what OS are you on?

Don't know yet why it wants to change it on your computer but not on Travis.

Comment (mellogarrett) [reply Textfield/Textfield.jsx]: I am using the Prettier VS Code plugin, v: 0.24.0 on MacOS Sierra (10.12.6)

Comment (filoozom) [reply Textfield/Textfield.jsx]: Haven't tried the plugin yet, maybe that's the issue?

Comment (filoozom) [reply Textfield/Textfield.jsx]: We actually don't have a ```prettier``` config file. The VS Code  plugin is looking for those :
- package.json - prettier key
- .prettierrc
- .prettier.config.js

Comment (mellogarrett) [reply Textfield/Textfield.jsx]: Hmm. How do we move forward?

Comment (filoozom) [reply Textfield/Textfield.jsx]: I guess you could still try disabling the plugin or running ```npm run prettier:write``` with VS Code closed, and if that works we might need to add a config file.

Comment (mellogarrett) [reply Textfield/Textfield.jsx]: I ran `npm i` to install Prettier locally in the project. Saved it and it fixed the formatting discrepancy.

**Pull request merged**